server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -984,6 +985,7 @@ public IndexSettings(final IndexMetadata indexMetadata, final Settings nodeSetti
          * Now this sortField (IndexSort) is stored in SegmentInfo and we need to maintain backward compatibility for them.
          */
         widenIndexSortType = IndexMetadata.SETTING_INDEX_VERSION_CREATED.get(settings).before(V_2_7_0);
+        assignedOnRemoteNode = RemoteStoreNodeAttribute.isRemoteDataAttributePresent(this.getNodeSettings());
```

Comment:
`NodeSettings` can't be changed without process restart, right?

Reply:
Yes, that is only populated from yml only .